oauthproxy.go

```diff
@@ -475,6 +484,9 @@ func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error)
 	}
 
 	redirect = req.Form.Get("rd")
+	if strings.HasPrefix(redirect, "/") {
```

Comment:
What's the reasoning behind this change? Struggling to tie it into the rest of the PR

Reply:
This is a crucial step to preserve the hostname during the oauth2 redirect. Up until now the oauth2 callback redirect was a server-relative redirect, so basically the user remained on the host of the oauth2 callback URL.